chain/types/message_receipt.go

```diff
@@ -3,15 +3,18 @@ package types
 import (
 	"bytes"
 
+	"github.com/ipfs/go-cid"
+
 	"github.com/filecoin-project/go-state-types/exitcode"
 )
 
 type MessageReceipt struct {
-	ExitCode exitcode.ExitCode
-	Return   []byte
-	GasUsed  int64
+	ExitCode   exitcode.ExitCode
+	Return     []byte
+	GasUsed    int64
+	EventsRoot cid.Cid // Root of Event AMT
```

Comment:
This will have to be a `*cid.Cid` because the FVM will return an `Option<Cid>`, with `None` meaning that no events were emitted. Unfortunately we can't use `cid.Undef` as a sentinel value for absence, because it fails to serialize in DAG-CBOR.

Reply:
Will push that change.